lib/custom/src/MShop/Service/Provider/Payment/PaypalPlus.php

```diff
@@ -0,0 +1,354 @@
+			'code' => 'payment.cvv',
+			'internalcode'=> 'cvv',
+			'label'=> 'Verification number',
+			'type'=> 'number',
+			'internaltype'=> 'integer',
+			'default'=> '',
+			'required'=> true
+		),
+		'payment.expirymonth' => array(
+			'code' => 'payment.expirymonth',
+			'internalcode'=> 'expiryMonth',
+			'label'=> 'Expiry month',
+			'type'=> 'select',
+			'internaltype'=> 'integer',
+			'default'=> '',
+			'required'=> true
+		),
+		'payment.expiryyear' => array(
+			'code' => 'payment.expiryyear',
+			'internalcode'=> 'expiryYear',
+			'label'=> 'Expiry year',
+			'type'=> 'select',
+			'internaltype'=> 'integer',
+			'default'=> '',
+			'required'=> true
+		),
+		'payment.company' => array(
+			'code' => 'payment.company',
+			'internalcode'=> 'company',
+			'label'=> 'Company',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.address1' => array(
+			'code' => 'payment.address1',
+			'internalcode'=> 'billingAddress1',
+			'label'=> 'Street',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.address2' => array(
+			'code' => 'payment.address2',
+			'internalcode'=> 'billingAddress2',
+			'label'=> 'Additional',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.city' => array(
+			'code' => 'payment.city',
+			'internalcode'=> 'billingCity',
+			'label'=> 'City',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.postal' => array(
+			'code' => 'payment.postal',
+			'internalcode'=> 'billingPostcode',
+			'label'=> 'Zip code',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.state' => array(
+			'code' => 'payment.state',
+			'internalcode'=> 'billingState',
+			'label'=> 'State',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.countryid' => array(
+			'code' => 'payment.countryid',
+			'internalcode'=> 'billingCountry',
+			'label'=> 'Country',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.telephone' => array(
+			'code' => 'payment.telephone',
+			'internalcode'=> 'billingPhone',
+			'label'=> 'Telephone',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+		'payment.email' => array(
+			'code' => 'payment.email',
+			'internalcode'=> 'email',
+			'label'=> 'E-Mail',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> false,
+			'public' => false,
+		),
+	);
+
+	private $provider;
+	private $basket;
```

Comment:
This variable is unused and can be removed

Reply:
thanks for note ,I have updated my code